ctrl+k cuts line into clipboard, ctrl+c clears clipboard if no selection

aerth opened the pull request:
ctrl+k cut multiple lines into clipboard

aerth commented:
This causes a panic when deleting nonexistent lines.

zyedidia commented:
I think that was actually a bug with `SelectLine`, see 0fac02cd08fe15f77d288185044d5aa6f16807da.

aerth commented:
0fac02c didn't fix it. Now there is a different panic when opening and pressing ctrl+k in an empty file.

```
Micro encountered an error: Operation not permitted on empty rope
*errors.errorString Operation not permitted on empty rope
/usr/local/go/src/runtime/panic.go:443 (0x45e5a9)
	gopanic: reflectcall(nil, unsafe.Pointer(d.fn), deferArgs(d), uint32(d.siz), uint32(d.siz))
/home/glenda/goforc/src/github.com/vinzmay/go-rope/rope.go:27 (0x48f99e)
	(*Rope).panicIfNil: panic(fmt.Sprintf("Operation not permitted on empty rope"))
/home/glenda/goforc/src/github.com/vinzmay/go-rope/rope.go:165 (0x4908ec)
	(*Rope).Split: rope.panicIfNil()
/home/glenda/goforc/src/github.com/vinzmay/go-rope/rope.go:190 (0x490ed2)
	(*Rope).Delete: _, r4 := r2.Split(length)
/home/glenda/goforc/src/github.com/aerth/micro/cmd/micro/buffer.go:124 (0x40180b)
	(*Buffer).Remove: b.r = b.r.Delete(start, end-start)
/home/glenda/goforc/src/github.com/aerth/micro/cmd/micro/eventhandler.go:33 (0x4072c1)
	ExecuteTextEvent: t.text = t.buf.Remove(t.start, t.end)
/home/glenda/goforc/src/github.com/aerth/micro/cmd/micro/eventhandler.go:98 (0x40781f)
	(*EventHandler).Execute: ExecuteTextEvent(t)
/home/glenda/goforc/src/github.com/aerth/micro/cmd/micro/eventhandler.go:83 (0x40763a)
	(*EventHandler).Remove: eh.Execute(e)
/home/glenda/goforc/src/github.com/aerth/micro/cmd/micro/cursor.go:95 (0x405749)
	(*Cursor).DeleteSelection: c.v.eh.Remove(c.curSelection[0], c.curSelection[1])
/home/glenda/goforc/src/github.com/aerth/micro/cmd/micro/view.go:553 (0x4299c9)
	(*View).HandleEvent: v.cursor.DeleteSelection()
/home/glenda/goforc/src/github.com/aerth/micro/cmd/micro/micro.go:256 (0x40ea25)
	main: view.HandleEvent(event)
/usr/local/go/src/runtime/proc.go:188 (0x4600c0)
	main: main_main()
/usr/local/go/src/runtime/asm_amd64.s:1998 (0x48e821)
	goexit: BYTE	$0x90	// NOP

```

aerth commented:
I am using Go 1.6.2 btw
